libcloud/compute/drivers/maxihost.py

```diff
@@ -0,0 +1,218 @@
+
+        if data['power_status']:
+            state = NodeState.RUNNING
+        else:
+            state = NodeState.STOPPED
+
+        for key in data:
+            extra[key] = data[key]
+
+        node = Node(id=data['id'], name=data['description'], state=state,
+                    private_ips=private_ips, public_ips=public_ips,
+                    driver=self, extra=extra)
+        return node
+
+    def list_locations(self, available=True):
+        """
+        List locations
+
+        If available is True, show only locations which are available
+        """
+        locations = []
+        data = self.connection.request('/regions')
+        for location in data.object['regions']:
+            if available:
+                if location.get('available'):
+                    locations.append(self._to_location(location))
+            else:
+                locations.append(self._to_location(location))
+        return locations
+
+    def _to_location(self, data):
+        name = data.get('location').get('city', '')
+        country = data.get('location').get('country', '')
+        return NodeLocation(id=data['slug'], name=name, country=country,
+                            driver=self)
+
+    def list_sizes(self):
+        """
+        List sizes
+        """
+        sizes = []
+        data = self.connection.request('/plans')
+        for size in data.object['servers']:
+            sizes.append(self._to_size(size))
+        return sizes
+
+    def _to_size(self, data):
+        extra = {'specs': data['specs'],
+                 'regions': data['regions'],
+                 'pricing': data['pricing']}
+        ram = data['specs']['memory']['total']
+        ram = re.sub("[^0-9]", "", ram)
+        return NodeSize(id=data['slug'], name=data['name'], ram=int(ram),
+                        disk=None, bandwidth=None,
+                        price=data['pricing']['usd_month'],
+                        driver=self, extra=extra)
+
+    def list_images(self):
+        """
+        List images
+        """
+        images = []
+        data = self.connection.request('/plans/operating-systems')
+        for image in data.object['operating-systems']:
+            images.append(self._to_image(image))
+        return images
+
+    def _to_image(self, data):
+        extra = {'operating_system': data['operating_system'],
+                 'distro': data['distro'],
+                 'version': data['version'],
+                 'pricing': data['pricing']}
+        return NodeImage(id=data['slug'], name=data['name'], driver=self,
+                         extra=extra)
+
+    def list_key_pairs(self):
+        """
+        List all the available SSH keys.
+
+        :return: Available SSH keys.
+        :rtype: ``list`` of :class:`KeyPair`
+        """
+        data = self.connection.request('/account/keys')
+        return list(map(self._to_key_pair, data.object['ssh_keys']))
+
+    def create_key_pair(self, name, public_key=''):
```

Comment:
Should remove default argument value if the argument is required (aka `(self, name, public_key)``).